### DISCUSSION TITLE
Clear version definition

### DISCUSSION THREAD

**Discussion by rhertogh** [yiisoft/yii2#20138] [Ideas]
Hi all,

Lately there has been a lot discussion and confusion about Yii2 versioning and PHP version compatibility. While it's good to have discussions (and a lot of valid points were raised) having confusion is not. Therefore I would propose to create a single and definitive version definition located within the Yii2 source code (e.g. in the [readme.md](https://github.com/yiisoft/yii2/blob/master/README.md)).

Based on the current developments and aforementioned discussions I would propose the following versioning:

| Yii2 Version | PHP version    | Development status                | EOL ¹                                                          |
|--------------|----------------|-----------------------------------|----------------------------------------------------------------|
| <= 2.0.49.*  | >= 5.4, <= 8.3 | security fixes only               | 23 Nov 2026 ³                                                  |
| >= 2.0.50    | >= 7.3, <= 8.4 | bug fixes and security fixes only | bugfixes till 23 Nov 2026 ³, security fixes till 21 Nov 2027 ⁴ |
| >= 2.2.0 ²   | >= 8.1         | active development                |                                                                |

¹ All mentioned dates may be subject to change and no rights can be derived from them.  
² Note: Yii 2.1 was [skipped](https://github.com/yiisoft/yii2/discussions/19831#discussioncomment-5858046), [Yii 2.2](https://github.com/yiisoft/yii2/tree/2.2) has not yet been released.  
³ [PHP 8.3 EOL date](https://www.php.net/supported-versions.php).  
⁴ [Expected PHP 8.4 EOL date](https://wiki.php.net/todo/php84).  

The table above is a first setup, all input is of course very welcome. 

---
Changelog:
| Date (UTC) | Change |
|--------|--------|
| 2024-03-23 16:44 | Initial version 
| 2024-03-23 22:34 | Changed maximum PHP version to 8.4 for Yii 2.0.50 and updated EOL accordingly | 
| 2024-03-27 20:28 | Added dates disclaimer, added note about Yii 2.1 |

**Comment by rhertogh** [comment]
@samdark @bizley @terabytesoftw @rob006 @mtangoo @schmunk42 and other @yiisoft/core-developers looking forward to hear your input.

**Reply by terabytesoftw** [reply]
I would just add, in the PHP 8.4 box for 2.0.50, which will already be available.

Excellent 100% agree.

**Reply by rhertogh** [reply]
@terabytesoftw I've updated the table. Since PHP 8.4 would push the EOL further back I've split the bugfixes and security fixes dates.

**Reply by bizley** [reply]
LGFM

**Reply by mtangoo** [reply]
Great Idea.  Will clear a lot of issues regarding versions

**Comment by mtangoo** [comment]
Just addition thing is clarifying that v2.1 never happened as some get confused with a leap from 2.0.x to 2.2.x

**Reply by rhertogh** [reply]
Good point, added in the footnotes with link

**Comment by rhertogh** [comment]
Pull Request: #20142

**Comment by samdark** [comment]
https://github.com/yiisoft-contrib/yiiframework.com/blob/master/config/release-cycle.php should be updated as well.

**Reply by rhertogh** [reply]
Nice page, didn't know about it.
I'll create a PR for it.

**Reply by rhertogh** [reply]
@samdark https://github.com/yiisoft-contrib/yiiframework.com/pull/1152
